packages/experiments/src/components/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,343 @@
+  private _theme: ITheme;
+  private _classNames: ISidebarClassNames;
+  private _colors: SidebarColors;
+  private _buttonStyles: IButtonStyles;
+
+  constructor(props: ISidebarProps) {
+    super(props);
+
+    this.state = {
+      isCollapsed: false
+    };
+  }
+
+  /*
+     * ComponentDidMount is used in the sidebar to adjust the height of the sidebar content
+     * to allow the content scrollbar to be correctly sized.
+     */
+  public componentDidMount(): void {
+    const parentId = this.props.id ? this.props.id : '';
+    const sidebar = document.getElementsByClassName(`ba-Sidebar-${parentId}`)[0] as HTMLElement;
+    const sidebarFooter = document.getElementsByClassName(`ba-SidebarFooter-${parentId}`)[0] as HTMLElement;
+    const sidebarContent = document.getElementsByClassName(`ba-SidebarContent-${parentId}`)[0] as HTMLElement;
+    if (sidebarContent && sidebar && sidebarFooter) {
+      sidebarContent.setAttribute('style', 'height: ' + (sidebar.offsetHeight - sidebarFooter.offsetHeight + 'px'));
+    }
+  }
+
+  public render(): JSX.Element {
+    const { theme, styles, collapseButtonStyles, className, collapseButtonAriaLabel, footerItems, id, items } = this.props;
+
+    this._theme = theme!;
+    this._colors = this.props.colors ? this.props.colors : SidebarColors.Light;
+    this._buttonStyles = getButtonColoredStyles(theme!, this._colors, this.props.buttonStyles);
+
+    this._classNames = getSidebarClassNames(getSidebarStyles(theme!, this._colors, styles), className, this.state.isCollapsed);
+
+    const ButtonAs = this._getButtonAs();
+
+    return (
+      <div className={this._classNames.root} role="menu" aria-orientation={'vertical'} aria-expanded={!this.state.isCollapsed}>
+        <ScrollablePane
+          className={this._classNames.content}
+          styles={{
+            contentContainer: {
+              overflowX: 'hidden'
+            }
+          }}
+        >
+          {this.props.collapsible && (
+            <ButtonAs
+              key={'baSidebarCollapsibleButton'}
+              iconProps={{ iconName: 'GlobalNavButton' }}
+              onClick={this.toggleCollapsed}
+              ariaLabel={collapseButtonAriaLabel}
+              theme={this._theme}
+              aria-expanded={!this.state.isCollapsed}
+              styles={concatStyleSets(this._buttonStyles, collapseButtonStyles)}
+            />
+          )}
+          <FocusZone direction={FocusZoneDirection.vertical}>
+            {items && items.map((item: ISidebarItemProps) => this._renderItemInSidebar(item))}
+          </FocusZone>
+        </ScrollablePane>
+        {footerItems && (
+          <FocusZone direction={FocusZoneDirection.vertical} className={this._classNames.footer} key={`baSidebarFooter${id}`}>
+            {footerItems.map((item: ISidebarItemProps) => this._renderItemInSidebar(item))}
+          </FocusZone>
+        )}
+      </div>
+    );
+  }
+
+  @autobind
```

Comment:
TIL

Reply:
heh...yeah. We used to do autobinds all over the place. We since moved to lambdas. This'll be part of the work needed to get the component ready to launch.